Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.cpp

```diff
@@ -487,20 +479,45 @@ EncoderId videoEncoderFindForFormat(WebKitVideoEncoder* self, const GRefPtr<GstC
     return candidates[0].first;
 }
 
-bool videoEncoderSupportsFormat(WebKitVideoEncoder* self, const GRefPtr<GstCaps>& caps)
+EncoderId videoEncoderFindForCodec(WebKitVideoEncoder* self, const String& codecName)
 {
-    return videoEncoderFindForFormat(self, caps) != None;
+    const char* gstCodec = nullptr;
+    if (codecName == "vp8"_s || codecName == "vp08"_s)
+        gstCodec = "vp8";
+    else if (codecName.startsWith("vp9"_s) || codecName.startsWith("vp09"_s))
+        gstCodec = "vp9";
+    else if (codecName.startsWith("avc1"_s))
+        gstCodec = "h264";
+    else if (codecName.startsWith("hvc1"_s) || codecName.startsWith("hev1"_s))
+        gstCodec = "h265";
+    else if (codecName.startsWith("av01"_s))
+        gstCodec = "av1";
+
+    if (!gstCodec)
+        return None;
+
+    auto name = makeString("video/x-"_s, gstCodec);
+    auto caps = adoptGRef(gst_caps_new_empty_simple(name.ascii().data()));
+    return videoEncoderFindForFormat(self, caps);
 }
 
-bool videoEncoderSetFormat(WebKitVideoEncoder* self, GRefPtr<GstCaps>&& caps, const String& codecString)
+bool videoEncoderSupportsCodec(WebKitVideoEncoder* self, const String& codecName)
 {
-    auto encoderId = videoEncoderFindForFormat(self, caps);
+    return videoEncoderFindForCodec(self, codecName) != None;
+}
+
+bool videoEncoderSetCodec(WebKitVideoEncoder* self, const String& codecName, std::optional<IntSize> size, std::optional<double> frameRate)
+{
+    auto encoderId = videoEncoderFindForCodec(self, codecName);
     if (encoderId == None) {
-        GST_ERROR_OBJECT(self, "No encoder found for format %" GST_PTR_FORMAT, caps.get());
+        GST_ERROR_OBJECT(self, "No encoder found for codec %s", codecName.ascii().data());
         return false;
     }
 
-    return videoEncoderSetEncoder(self, encoderId, WTFMove(caps), codecString);
+    auto [inputCaps, outputCaps] = GStreamerCodecUtilities::capsFromCodecString(codecName, WTFMove(size), frameRate);
```

Comment:
Moving this parameter here without having it as a && in the parameter list is a bit weird. If the idea is moving inside the function, the most logical thing, IMHO, would be to move when calling this function.

Reply:
Hmm yes, the size param in capsFromCodecString is not `&&`, so moving it here doesn't make much sense i suppose. i can remove it.

Reply:
Well, moving it here makes sense, but it makes even more sense moving it in the function calling this one. Sorry if I didn't explain it properly. Anyway, not big thing.